src/NuGet.Core/NuGet.Protocol/EnhancedHttpRetryHelper.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Common;
+
+namespace NuGet.Protocol
+{
+    /// <summary>
+    /// Represents a helper class for determining if enhanced retry is enabled and what settings to use.
+    /// </summary>
+    internal class EnhancedHttpRetryHelper
+    {
+        /// <summary>
+        /// The default delay in milliseconds between retries.
+        /// </summary>
+        public const int DefaultDelayMilliseconds = 1000;
+
+        /// <summary>
+        /// The default value indicating whether or not enhanced HTTP retry is enabled.
+        /// </summary>
+        public const bool DefaultEnabled = false;
+
+        /// <summary>
+        /// The default number of times to retry.
+        /// </summary>
+        public const int DefaultRetryCount = 6;
+
+        /// <summary>
+        /// The environment variable used to change the delay value.
+        /// </summary>
+        public const string DelayInMillisecondsEnvironmentVariableName = "NUGET_ENHANCED_NETWORK_RETRY_DELAY_MILLISECONDS";
+
+        /// <summary>
+        /// The environment variable used to enable or disable the enhanced HTTP retry.
+        /// </summary>
+        public const string IsEnabledEnvironmentVariableName = "NUGET_ENABLE_ENHANCED_HTTP_RETRY";
+
+        /// <summary>
+        /// The environment variable used to change the retry value.
+        /// </summary>
+        public const string RetryCountEnvironmentVariableName = "NUGET_ENHANCED_MAX_NETWORK_TRY_COUNT";
+
+        private readonly IEnvironmentVariableReader _environmentVariableReader;
+
+        private bool? _isEnabled = null;
+
+        private int? _retryCount = null;
+
+        private int? _delayInMilliseconds = null;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EnhancedHttpRetryHelper" /> class.
+        /// </summary>
+        /// <param name="environmentVariableReader">A <see cref="IEnvironmentVariableReader" /> to use when reading environment variables.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="environmentVariableReader" /> is <c>null</c>.</exception>
+        public EnhancedHttpRetryHelper(IEnvironmentVariableReader environmentVariableReader)
+        {
+            _environmentVariableReader = environmentVariableReader ?? throw new ArgumentNullException(nameof(environmentVariableReader));
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether or not enhanced HTTP retry should be enabled.  The default value is <c>true</c>.
```

Comment:
The comment says the default value is true but line line no. 22 says `public const bool DefaultEnabled = false;`. It is possible that I am missing something.

Reply:
Thanks, I've fixed this!